src/briefcase/integrations/docker.py

```diff
@@ -504,8 +504,8 @@ def _dockerize_path(self, arg: str) -> str:  # pragma: no-cover-if-is-windows
         """
         if arg == sys.executable:
             return f"python{self.python_version}"
-        arg = arg.replace(os.fsdecode(self.host_bundle_path), "/app")
-        arg = arg.replace(os.fsdecode(self.host_data_path), str(self.docker_data_path))
```

Comment:
I know we've touched on `os.fsdecode()` before but I wanted to make sure we have the same expectations for its use.

As stated before, it is safest to use `os.fsdecode()` to cast an arbitrary object to a string for use as a file path since some objects that have a `__str__()` will never be valid file paths. (It also manages special escaping for its inverse function `os.fsencode()`.)

When using `os.fsdecode()` on an object known to be a `Path`, it is really just a long way of calling `str()`. That is, `fsdecode(obj)` calls `fspath(obj)` calls `type(obj).__fspath__(obj)` which is just `str(self)` for `Path`.

So...do we want to use `os.fsdecode()` on objects that are known to be an instance of `Path`? And is my understanding correct?

Reply:
If we're absolutely 100% certain it's a `Path`, and won't ever be `None`, then yes - `str()` is safe. However, using `os.fsdecode()` regardless is one of those safety mechanisms by practice - if we use it even when we *strictly* don't have to, we're more likely to remember we have to do it when it *does* matter.

Reply:
Yeah; that's mostly my thought as well - it's safer to just use `fsdecode()`....although, there's so many exceptions to this rule currently....namely all the f-strings implicitly converting Paths to strings. ehh...it is what it is, though.

Reply:
Since I git amended, I'll clarify here that I put the `os.fsdecode()` calls back.